
# HG changeset patch
# User thartmann
# Date 1596694256 -7200
# Node ID b83c399f01c1e350196b38bfccc8814c22a05e93
# Parent  4594eb2f1f30bfaf897b812fb9d78ced7efab203
8250861: Crash in MinINode::Ideal(PhaseGVN*, bool)
Summary: Added missing NULL checks.
Reviewed-by: kvn, chagedorn

diff -r 4594eb2f1f30 -r b83c399f01c1 src/hotspot/share/opto/addnode.cpp
--- a/src/hotspot/share/opto/addnode.cpp	Tue Aug 11 13:13:54 2020 +0000
+++ b/src/hotspot/share/opto/addnode.cpp	Thu Aug 06 08:10:56 2020 +0200
@@ -1007,7 +1007,7 @@
 
     // Transform MIN2(x + c0, MIN2(x + c1, z)) into MIN2(x + MIN2(c0, c1), z)
     // if x == y and the additions can't overflow.
-    if (phase->eqv(x,y) &&
+    if (phase->eqv(x,y) && tx != NULL &&
         !can_overflow(tx, x_off) &&
         !can_overflow(tx, y_off)) {
       return new MinINode(phase->transform(new AddINode(x, phase->intcon(MIN2(x_off, y_off)))), r->in(2));
@@ -1015,7 +1015,7 @@
   } else {
     // Transform MIN2(x + c0, y + c1) into x + MIN2(c0, c1)
     // if x == y and the additions can't overflow.
-    if (phase->eqv(x,y) &&
+    if (phase->eqv(x,y) && tx != NULL &&
         !can_overflow(tx, x_off) &&
         !can_overflow(tx, y_off)) {
       return new AddINode(x,phase->intcon(MIN2(x_off,y_off)));

